### PR TITLE
Chore/prettify and lint

### PR DESCRIPTION
**Pull request by DanielaValero**
**What kind of change does this PR introduce?**
* Build feature: Uses the webpack eslint config. Edits some settings, that make sense to have in the cli. Example no warning when we use console.log
* Uses `eslint --fix` to beautify code. In webpack they are using `beautify-lint` however it seems to me that it does the same than `eslint --fix`, therefore left only the one of `eslint`
* Chore: Fixes all lintin errors in the code base
* Updates all the dependencies
* Removed `fix-commit-js` as it is not really working

**note** Wanted to use prettier for the automatic prettification of the code, but was quite complicated. More details here: https://github.com/webpack/webpack-cli/pull/102#issuecomment-316046523

**Did you add tests for your changes?**
N/A

**If relevant, did you update the documentation?**
N/A

**Does this PR introduce a breaking change?**

Nope

### CODE REVIEW

**Comment by DanielaValero** [comment]
Hi guys, 

I noticed the broken tests, which are also broken in `master`. If apart from that, the PR is good for you, I'd suggest to get it merged. As it touches many files, the longer we take to merge it, the more merge conflicts we'll get here.

I will start right away a RP with the fixes for the broken tests.

File: package.json [hunk 1]
```diff
@@ -16,65 +16,62 @@
...
     "pre-commit": "^1.2.2"
-  },
-  "lint-staged": {
-    "*.js": [
-      "eslint --fix",
-      "git add"
-    ]
-  },
-  "pre-commit": "lint:staged",
-  "jest": {
```
Comment (okonet) [inline package.json]: Did you drop this intentionally?

File: package.json [hunk 2]
```diff
@@ -16,65 +16,62 @@
...
-    "lint-staged": "^3.2.8",
+    "ajv": "^5.2.2",
+    "ajv-keywords": "^2.1.0",
+    "babel-cli": "^6.24.1",
+    "babel-eslint": "^7.2.3",
+    "babel-jest": "^20.0.3",
+    "babel-polyfill": "^6.23.0",
+    "eslint": "^4.2.0",
+    "eslint-plugin-node": "^5.1.0",
+    "eslint-plugin-prettier": "^2.1.2",
```
Comment (okonet) [inline package.json]: I see this dependency but I could not find the reference to it anywhere. Did you forget to add it to `eslintrc`?

File: package.json (hunk 1 above)
Comment (DanielaValero) [inline package.json]: Oh, good finding! did not want to drop it, was in the merge conflict solving, etc, etc

File: package.json (hunk 2 above)
Comment (DanielaValero) [inline package.json]: I forgot to remove it, I had added it when I was trying to add prettier to automatically style our code, but after I figured out that prettier is not yet there, removed it, and forgot to get it out from the dependencies.

**Comment by DanielaValero** [comment]
@okonet added your comments, it's ready for a second look

**Comment by ev1stensberg** [comment]
Could you rebase against master, as the snapshot tests are making the CI fail?

**Comment by DanielaValero** [comment]
Hi @ev1stensberg 

The branch is already up-to-date, the tests are also failing in the master branch. Once this is merged, I'll check them out, to make a PR fixing them

**Comment by ev1stensberg** [comment]
Gotcha ✌️

**Comment by DanielaValero** [comment]
Hello @ev1stensberg @okonet As there is no more comment on this PR, would it be possible to merge it?

Otherwise, let me know what the blocker issue is, so I can apply it, and get this in asap. 

There are many touched files, and the longer we wait to merge it, the more extra not necessary work will be raised

**Comment by ev1stensberg** [comment]
Haven't had a chance to test this, could it chill until tomorrow?

**Comment by DanielaValero** [comment]
@ev1stensberg sure! :) Thanks for taking the time

**Comment by ev1stensberg** [comment]
Could you do `rm -rf node_modules && npm install` ? Otherwise, lgtm

**Comment by ematipico** [comment]
Would it be possible to merge the PR? I'd like to have it merged as I am working on implementing `Flow`. This could generate conflicts and since this PR has been already reviewed, I'd like to rebase on it. Cheers!

**Comment by okonet** [comment]
Some tests are failing for this PR. I'm not sure if it's because of those changes or not.

**Comment by okonet** [comment]
I'll look at test and merge this one.

File: lib/creator/transformations/context/context.js [hunk 3]
```diff
@@ -13,9 +14,18 @@ const utils = require('../../../transformations/utils');
...
-		.filter(p => utils.isAssignment(j, p, utils.pushCreateProperty, 'context', webpackProperties));
+	if (webpackProperties) {
+		return ast
+			.find(j.ObjectExpression)
+			.filter(p =>
+				utils.isAssignment(
+					j,
+					p,
+					utils.pushCreateProperty,
+					"context",
```
Comment (okonet) [inline lib/creator/transformations/context/context.js]: So FYI this is what was breaking the test.

File: lib/creator/transformations/context/context.test.js [hunk 4]
```diff
@@ -1,5 +1,7 @@
-const defineTest = require('../../../transformations/defineTest');
+"use strict";
 
-defineTest(__dirname, 'context', 'context-0', 'path.resolve(__dirname, "app")');
-defineTest(__dirname, 'context', 'context-1', '\'./some/fake/path\'');
-defineTest(__dirname, 'context', 'context-2', 'contextVariable');
+const defineTest = require("../../../transformations/defineTest");
+
+defineTest(__dirname, "context", "context-0", "path.resolve(__dirname, 'app')");
```
Comment (okonet) [inline lib/creator/transformations/context/context.test.js]: This one is to blame!

**Pull request merged**